rbd/pool_nautilus.go

```diff
@@ -23,9 +23,11 @@ import (
 // Implements:
 //  int rbd_pool_metadata_get(rados_ioctx_t io_ctx, const char *key, char *value, size_t *val_len);
 func GetPoolMetadata(ioctx *rados.IOContext, key string) (string, error) {
+	if ioctx == nil {
+		return "", ErrNoIOContext
+	}
 	cKey := C.CString(key)
 	defer C.free(unsafe.Pointer(cKey))
-
```

Comment:
Very minor nit: You lost some whitespace here. I'm not dead set on having it but I like a liine of whitespace here to make a visible separation between the "preamble" to the meat of the function.